src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -0,0 +1,607 @@
+        auto axis_p = pattern::wrap_type<opset8::Constant>();
+        auto pattern_root = pattern::wrap_type<opset8::Gather>({pattern::any_input(), indices_p, axis_p});
+
+        auto callback = [=](pattern::Matcher& m) {
+            const auto & pattern_map = m.get_pattern_map();
+            const auto & output = pattern_map.at(pattern_root);
+
+            auto order = ov::get_constant_from_source(pattern_map.at(indices_p));
+            auto axis = ov::get_constant_from_source(pattern_map.at(axis_p));
+            if (!order || !axis) {
+                return false;
+            }
+
+            ric_attr::init(output, order->cast_vector<int64_t>(), axis->cast_vector<int64_t>().at(0));
+            return true;
+        };
+
+        auto m = std::make_shared<pattern::Matcher>(pattern_root, matcher_name);
+        register_matcher(m, callback);
+    }
+};
+}// namespace init
+
+namespace prop {
+namespace {
+std::shared_ptr<opset8::Constant> create_const(const std::vector<int64_t> & values) {
+    return opset8::Constant::create(ov::element::i64, ov::Shape{values.size()}, values);
+}
+}// namespace
+
+class Binary : public ngraph::pass::MatcherPass {
+public:
+    Binary() {
+        MATCHER_SCOPE(Binary);
+        auto pattern_root = pattern::wrap_type<op::util::BinaryElementwiseArithmetic, opset8::FakeQuantize>();
+
+        auto callback = [=](pattern::Matcher& m) {
+            const auto & root = m.get_match_root();
+            const auto & inputs = root->inputs();
+
+            std::map<size_t, ric_attr::Attribute> attrs;
+            for (const auto & input : inputs) {
+                auto output = input.get_source_output();
+                if (ric_attr::has(output)) {
+                    attrs.insert({input.get_index(), ric_attr::get(output).propagate()});
+                } else if (!ov::is_type<opset8::Constant>(output.get_node())) {
+                    // If number of non-constant inputs and without RIC attr is greater than 0 we have to skip
+                    // propagation because it is not efficient to have a lot of RIC copies on data path.
+                    return false;
+                }
+            }
+
+            if (attrs.empty()) return false;
+
+            // Check that all RIC attrs can be merged and then merge them
+            auto ric = attrs.begin()->second;
+            auto rank = root->get_input_partial_shape(attrs.begin()->first).rank();
+            if (rank.is_dynamic()) return false;
+            auto data_rank = rank.get_length();
+
+            for (const auto & item : attrs) {
+                const auto & input_rank = root->get_input_partial_shape(item.first).rank();
+                if (input_rank.is_static() &&
+                    input_rank.get_length() == data_rank &&
+                    ric.can_be_merged_with(item.second) ) {
+                    ric.merge_with(item.second);
+                } else {
+                    return false;
```

Comment:
should this be tested?

Reply:
Due to not big amount supported ops I can't simulate such case so far.